proto/v1/instance_service.proto

```diff
@@ -0,0 +1,174 @@
+    option (google.api.http) = {
+      get: "/v2/{parent=environments/*}/instances"
+    };
+    option (google.api.method_signature) = "parent";
+  }
+
+  rpc CreateInstance(CreateInstanceRequest) returns (Instance) {
+    option (google.api.http) = {
+      post: "/v2/{parent=environments/*}/instances"
+      body: "instance"
+    };
+    option (google.api.method_signature) = "parent,instance";
+  }
+
+  rpc UpdateInstance(UpdateInstanceRequest) returns (Instance) {
+    option (google.api.http) = {
+      patch: "/v2/{instance.name=environments/*/instances/*}"
+      body: "instance"
+    };
+    option (google.api.method_signature) = "instance,update_mask";
+  }
+
+  rpc DeleteInstance(DeleteInstanceRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/v2/{name=environments/*/instances/*}"
+    };
+    option (google.api.method_signature) = "name";
+  }
+
+  rpc UndeleteInstance(UndeleteInstanceRequest) returns (Instance) {
+    option (google.api.http) = {
+      post: "/v2/{name=environments/*/instances/*}:undelete"
+      body: "*"
+    };
+  }
+}
+
+message GetInstanceRequest {
+  // The name of the instance to retrieve.
+  // Format: environments/{environment}/instances/{instance}
+  string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message ListInstancesRequest {
+  // The parent, which owns this collection of instances.
+  // Format: environments/{environment}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The maximum number of instances to return. The service may return fewer than
+  // this value.
+  // If unspecified, at most 50 instances will be returned.
+  // The maximum value is 1000; values above 1000 will be coerced to 1000.
+  int32 page_size = 2;
+
+  // A page token, received from a previous `ListInstances` call.
+  // Provide this to retrieve the subsequent page.
+  //
+  // When paginating, all other parameters provided to `ListInstances` must match
+  // the call that provided the page token.
+  string page_token = 3;
+
+  // Show deleted instances if specified.
+  bool show_deleted = 4;
+}
+
+message ListInstancesResponse {
+  // The instances from the specified publisher.
+  repeated Instance instances = 1;
+
+  // A token, which can be sent as `page_token` to retrieve the next page.
+  // If this field is omitted, there are no subsequent pages.
+  string next_page_token = 2;
+}
+
+message CreateInstanceRequest {
+  // The parent resource where this instance will be created.
+  // Format: environments/{environment}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The instance to create.
+  Instance instance = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // The ID to use for the instance, which will become the final component of
+  // the instance's resource name.
+  //
+  // This value should be 4-63 characters, and valid characters
+  // are /[a-z][0-9]-/.
+  string instance_id = 3;
```

Comment:
This is the place to provide the resource-id while creating a resource. @ecmadao